bson/src/main/org/bson/BsonBinary.java

```diff
@@ -165,6 +165,9 @@ public byte getType() {
     /**
      * Gets the data of this Binary.
      *
+     * This method returns the internal copy of the byte array, so only modify the contents of the returned array if the intention is to
```

Comment:
This is sort of the root of the problem, but we can't really change the behavior without risk of breaking applications.  So just documenting it.